### DISCUSSION TITLE
How can i read data async from buffer ?

### DISCUSSION THREAD

**Discussion by orcunor** [IntergatedCircuits/HidSharp#47] [Q&A]
How can I read the data from the buffer with asynchronously? 
I'm trying to make a simple win form app. I can read and write synchronously, but with which function or how can I do asynchronous reading with HidSharp ? Can anybody help me

**Comment by benedekkupper** [comment]
Please check HidSharp.Test/Program.cs, I guess the THREAD_POOL_RECEIVED_EVENT_APPROACH will get you closest to an async apach.
